lib/mxcreply.cpp

```diff
@@ -90,10 +91,17 @@ MxcReply::MxcReply()
 
 qint64 MxcReply::readData(char *data, qint64 maxSize)
 {
-    return d->m_device->read(data, maxSize);
+    const qint64 size = d->m_device->read(data, maxSize);
+    d->m_bytesRead += size;
+    return size;
 }
 
 void MxcReply::abort()
 {
     d->m_reply->abort();
 }
+
+qint64 Quotient::MxcReply::bytesAvailable() const
+{
+    return d->m_reply->bytesAvailable() + d->m_bytesRead;
```

Comment:
At least, I think we should be consistent and call to `m_device`, rather than `m_reply` that will be already read from in E2EE cases, right?
```suggestion
    return d->m_device->bytesAvailable() + d->m_bytesRead;
```